portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -2339,5 +2349,33 @@ RenderResult getInlineRenderingForTool(HttpServletResponse res, HttpServletReque
 		
 		return rval;
 	}
+	
+	/**
+	 * Checks if current user is being impersonated (via become user/sutool) and returns userEId of
+	 * the impersonator. Adapted from SkinnableLogin's isImpersonating()
+	 * @return userEid of impersonator, or empty string if not being impersonated
+	 */
+	private String getImpersonatorEid()
+	{
+		Session currentSession = SessionManager.getCurrentSession();
+		String currentUserId = currentSession.getUserId();
+		UsageSession originalSession = (UsageSession) currentSession.getAttribute(UsageSessionService.USAGE_SESSION_KEY);
+		String originalUserId = originalSession == null ? "" : originalSession.getUserId();
+		
+		if (originalSession != null && !originalUserId.equals(currentUserId))
+		{
+			try
+			{
+				User originalUser = UserDirectoryService.getUser(originalUserId);
+				return originalUser.getEid();
+			}
+			catch (UserNotDefinedException e)
+			{
+				return "";
```

Comment:
How about a log.debug here and then just let the return "" below return the value